sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileClient.java

```diff
@@ -334,25 +344,38 @@ public FileDownloadInfo downloadWithProperties() {
      *
      * <p>Download the file from 1024 to 2048 bytes with its metadata and properties and without the contentMD5. </p>
      *
-     * {@codesnippet com.azure.storage.file.FileClient.downloadWithPropertiesWithResponse#FileRange-Boolean-Duration-Context}
+     * {@codesnippet com.azure.storage.file.FileClient.downloadWithResponse#OutputStream-FileRange-Boolean-Duration-Context}
      *
      * <p>For more information, see the
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/get-file">Azure Docs</a>.</p>
      *
+     * @param stream A non-null {@link OutputStream} where the downloaded data will be written.
      * @param range Optional byte range which returns file data only from the specified range.
      * @param rangeGetContentMD5 Optional boolean which the service returns the MD5 hash for the range when it sets to
      * true, as long as the range is less than or equal to 4 MB in size.
      * @param timeout An optional timeout applied to the operation. If a response is not returned before the timeout
      * concludes a {@link RuntimeException} will be thrown.
      * @param context Additional context that is passed through the Http pipeline during the service call.
-     * @return A response containing the {@link FileDownloadInfo file download info} headers and response status code
+     * @return A response containing the headers and response status code
+     * @throws NullPointerException If {@code stream} is {@code null}.
      * @throws RuntimeException if the operation doesn't complete before the timeout concludes.
      */
-    public Response<FileDownloadInfo> downloadWithPropertiesWithResponse(FileRange range, Boolean rangeGetContentMD5,
+    public Response<Void> downloadWithResponse(OutputStream stream, FileRange range, Boolean rangeGetContentMD5,
         Duration timeout, Context context) {
-        Mono<Response<FileDownloadInfo>> response = fileAsyncClient
-            .downloadWithPropertiesWithResponse(range, rangeGetContentMD5, context);
-        return Utility.blockWithOptionalTimeout(response, timeout);
+        Objects.requireNonNull(stream, "'stream' cannot be null.");
+
+        Mono<Response<Void>> download = fileAsyncClient.downloadWithResponse(range, rangeGetContentMD5, context)
+            .flatMapMany(response -> response.getValue().doOnNext(buffer -> {
+                try {
+                    stream.write(FluxUtil.byteBufferToArray(buffer));
+                } catch (IOException ex) {
+                    throw logger.logExceptionAsError(new UncheckedIOException(ex));
+                }
+            }).map(buffer -> response))
+            .last()
```

Comment:
If `response.getValue()` emits `Flux.empty()` then `last()` operator can throw ` NoSuchElementException` Ref: https://projectreactor.io/docs/core/release/api/reactor/core/publisher/Flux.html#last--

Reply:
I guess we could simplify it a bit by using flatmap() and reduce() something like below (untested):

```java
Mono<Void> download = fileAsyncClient.downloadWithResponse(range, rangeGetContentMD5, context)
  .flatmap(response -> response.getValue().reduce(() -> stream, (stream, bb) -> { do write;, return stream; });
```

Reply:
Updated to use this